src/cmds/core/macro.py

```diff
@@ -0,0 +1,147 @@
+            macro: Macro = result.first()
+            if macro:
+                macro.text = text
+                await session.commit()
+                return await ctx.respond(f"Macro #{macro_id} has been updated.", ephemeral=True)
+            else:
+                return await ctx.respond(f"Macro #{macro_id} has not been found.", ephemeral=True)
+
+    @macro.command(description="List all macro's.")
+    @commands.cooldown(1, 30, commands.BucketType.user)
+    async def list(self, ctx: ApplicationContext) -> Interaction | WebhookMessage:
+        """List all macro's."""
+        async with AsyncSessionLocal() as session:
+            stmt = select(Macro)
+            result = await session.scalars(stmt)
+            macros: Sequence[Macro] = result.all()
+            # If no macros are returned, show a message.
+            if not macros:
+                return await ctx.respond("No macros have been added yet.")
+
+            embed = Embed(title="Macros", description="List of all macros.")
+            for macro in macros:
+                embed.add_field(name=f"Macro #{macro.id} - name: {macro.name}", value=macro.text, inline=False)
+            return await ctx.respond(embed=embed, ephemeral=True)
+
+    @macro.command(description="Send text from a macro by providing the name.")
+    @commands.cooldown(1, 30, commands.BucketType.user)
+    async def send(self,
+                   ctx: ApplicationContext,
+                   name: str,
+                   channel: GuildChannel = None
+                   ) -> Interaction | WebhookMessage:
+        """Send text from a macro by providing the name If a mod or higher, they can send to a remote channel."""
+        name = name.lower()
+        async with AsyncSessionLocal() as session:
+            stmt = select(Macro).filter(Macro.name == name)
+            result = await session.scalars(stmt)
+            macro: Macro = result.first()
+
+            if macro:
+                if channel:
+                    allowed_roles = {role.id for role in ctx.user.roles}
+                    required_roles = set(
+                        settings.role_groups.get("ALL_ADMINS", [])
+                        + settings.role_groups.get("ALL_HTB_STAFF", [])
+                        + settings.role_groups.get("ALL_MODS", [])
+                    )
+                    if allowed_roles & required_roles:
+                        await channel.send(f"{macro.text}")
+                        return await ctx.respond(f"Macro {name} has been sent to {channel.mention}.", ephemeral=True)
+                    return await ctx.respond("You don't have permission to send macros in other channels.",
+                                             ephemeral=True)
+                return await ctx.respond(f"{macro.text}")
+            return await ctx.respond(f"Macro #{name} has not been found. "
+                                     "Check the list of macros via the command `/macro list`.", ephemeral=True)
```

Comment:
If there is the capacity, it would make sense to return early on bad cases and return at the end for the happy path.
```
if not macro:
    return await ctx.respond(
        f"Macro #{name} has not been found.  Check the list of macros via the command `/macro list`.",
        ephemeral=True
)
```

Also, since channel is required why in function definition has a default value of None?

Reply:
@dimoschi - channel is not required as it's also possible to send it from a remote channel. I will get back on the other changes shortly.